contexts/chat.tsx

```diff
@@ -124,6 +130,21 @@ const ChatContextProvider: React.FC<ChatContextProps> = ({
   const threadInitialized = useRef(false);
   const [shouldRestart, setShouldRestart] = useState(false);
 
+  const switchToThread = async (
+    script: string,
+    id: string,
+    scriptId: string
+  ) => {
+    if (id !== thread) {
+      setScript(script);
+      setThread(id);
+      setScriptContent((await getScript(scriptId))?.script || []);
+      setScriptId(scriptId);
+      setForceRun(true);
```

Comment:
I think you shouldn't need to set`setForceRun`, setting `Thread` and `ShouldRestart` should be sufficient. I see a double refreshing issue when switch to thread, like the thread reloads twices on screen.

Reply:
So I thought the same thing and removed it in a previous commit, but actually had to revert that change, because it caused the issue where it just reused the chat from the previous thread

Reply:
I can try removing the shouldRestart, but I'm not sure if that will work

Reply:
Ok I fixed this by removing `setShouldRestart` and keeping `setForceRun` the reason I opted for this is because it doesn't look like `setShouldRestart` consistently triggers the `run` event (which updates the messages). It would probably be more ideal to get `shouldRestart` to update the messages, but in the interest of time spent on this issue I opted to use the tool already available